tcms/testplans/forms.py

```diff
@@ -188,7 +188,7 @@ def clean_parent(self):
         except TestPlan.DoesNotExist:
             raise forms.ValidationError('The plan does not exist in database.')
 
-    def populate(self, product_id):
+    def populate(self, product_id=None):
```

Comment:
Why is this necessary ?

Reply:
There are two places in `tcms.testplans.views` where we are using this function without passing a parameter (line `134` and `291`), so `pylint` throws an issue for that. Putting a default value for this parameter solves the problem.

Reply:
I can't get pylint to report this error but here are a few more comments:

```
    130         search_form = SearchPlanForm(request.GET)
    131         if request.GET.get('product'):
    132             search_form.populate(product_id=request.GET['product'])
    133         else:
    134             search_form.populate()
```

So line 134 uses SearchForm.populate, not BasePlanForm.populate. In addition the entire if statement above is useless. Just call
```
search_form.populate(product_id=request.GET.get('product'))
```

Same goes for line 291!